### PR TITLE
vim: don't assume all pythons are framework builds

### LINKED ISSUE #1165: vim: installation fails when using pyenv
# Please follow the general troubleshooting steps first:

- [x] Ran `brew update` and retried your prior step?
- [x] Ran `brew doctor`, fixed as many issues as possible and retried your prior step?
- [x] If you're seeing permission errors tried running `sudo chown -R $(whoami) $(brew --prefix)`?

### Bug reports:

This issue was originally reported to pyenv, here: https://github.com/yyuu/pyenv/issues/589

When running `brew install/upgrade vim` and pyenv, an error occurs when attempting to upgrade the package. This appears to be due to incompatible unicode formats. 

https://gist.github.com/d9f660ae9f77b8ae449335df283247be

### Steps to reproduce
Requirements:
- pyenv
- vim uninstalled

Install
```
pyenv global 2.7.11
brew install https://raw.githubusercontent.com/Homebrew/homebrew-core/e0de5d3e7bf9976ea9232b083add47fc9106e09e/Formula/vim.rb --with-python --with-cscope --with-lua --with-ruby --with-luajit --with-perl
```

Upgrade
```
pyenv global system
brew install https://raw.githubusercontent.com/Homebrew/homebrew-core/e0de5d3e7bf9976ea9232b083add47fc9106e09e/Formula/vim.rb --with-python --with-cscope --with-lua --with-ruby --with-luajit --with-perl
pyenv global 2.7.11
brew upgrade vim
```

Not sure what the best approach here is. Checking to see if pyenv is available seems hacky, but workable.

### PR DESCRIPTION
**Pull request by tdsmith**
Fixes #1165.

### CODE REVIEW

**Comment by tdsmith** [comment]
Applies https://github.com/vim/vim/pull/980.

File: Formula/vim.rb [hunk 1]
```diff
@@ -61,6 +61,13 @@ def install
       ENV.append "LDFLAGS", "-F#{buildpath}/Frameworks"
     end
 
+    # Fix build with flat Python builds
+    # https://github.com/Homebrew/homebrew-core/issues/1165
+    # https://github.com/vim/vim/pull/980
+    inreplace "src/auto/configure",
```
Comment (MikeMcQuaid) [inline Formula/vim.rb]: Presumably we can't apply https://github.com/vim/vim/pull/980.patch?

File: Formula/vim.rb [hunk 2]
```diff
@@ -61,6 +61,13 @@ def install
       ENV.append "LDFLAGS", "-F#{buildpath}/Frameworks"
     end
 
+    # Fix build with flat Python builds
```
Comment (MikeMcQuaid) [inline Formula/vim.rb]: `flat (non-framework)` just to be more obvious?

**Comment by MikeMcQuaid** [comment]
One question and a nit then 👍

**Comment by tdsmith** [comment]
The PR was pulled and vim does continuous releases so we can just bump the version 🎉

**Comment by MikeMcQuaid** [comment]
Oh nice 🎉

**Pull request closed without merging**